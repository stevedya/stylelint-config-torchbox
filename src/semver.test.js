const path = require('path');
const stylelint = require('stylelint');

const pkg = require('../package.json');

describe('semver - should those tests break, consider releasing a new major version of the package', () => {
    it('dependencies', () => {
        expect(pkg.dependencies).toMatchInlineSnapshot(`
            {
              "stylelint-config-prettier": "^9.0.3",
              "stylelint-config-standard": "^24.0.0",
              "stylelint-declaration-strict-value": "^1.8.0",
              "stylelint-order": "^5.0.0",
              "stylelint-scss": "^4.0.0",
            }
        `);
    });

    it('peerDependencies', () => {
        expect(pkg.peerDependencies).toMatchInlineSnapshot(`
            {
              "stylelint": "^14.0.0",
            }
        `);
    });

    it('config contents', async () => {
        const config = await stylelint.resolveConfig(__filename, {
            configFile: path.join(__dirname, '..', 'config.js'),
        });
        expect(config.rules).toMatchSnapshot();
    });
});
