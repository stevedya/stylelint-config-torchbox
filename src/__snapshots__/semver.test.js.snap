// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`semver - should those tests break, consider releasing a new major version of the package config contents 1`] = `
{
  "alpha-value-notation": null,
  "at-rule-empty-line-before": null,
  "at-rule-name-case": null,
  "at-rule-name-newline-after": null,
  "at-rule-name-space-after": null,
  "at-rule-no-unknown": null,
  "at-rule-no-vendor-prefix": [
    true,
  ],
  "at-rule-semicolon-newline-after": null,
  "at-rule-semicolon-space-before": null,
  "block-closing-brace-empty-line-before": null,
  "block-closing-brace-newline-after": null,
  "block-closing-brace-newline-before": null,
  "block-closing-brace-space-after": null,
  "block-closing-brace-space-before": null,
  "block-no-empty": [
    true,
  ],
  "block-opening-brace-newline-after": null,
  "block-opening-brace-newline-before": null,
  "block-opening-brace-space-after": null,
  "block-opening-brace-space-before": null,
  "color-function-notation": null,
  "color-hex-case": null,
  "color-hex-length": [
    "short",
  ],
  "color-named": [
    "never",
  ],
  "color-no-invalid-hex": [
    true,
  ],
  "comment-empty-line-before": [
    "always",
    {
      "except": [
        "first-nested",
      ],
      "ignore": [
        "stylelint-commands",
      ],
    },
  ],
  "comment-no-empty": [
    true,
  ],
  "comment-whitespace-inside": [
    "always",
  ],
  "custom-media-pattern": null,
  "custom-property-empty-line-before": [
    "always",
    {
      "except": [
        "after-custom-property",
        "first-nested",
      ],
      "ignore": [
        "after-comment",
        "inside-single-line-block",
      ],
    },
  ],
  "custom-property-no-missing-var-function": [
    true,
  ],
  "custom-property-pattern": null,
  "declaration-bang-space-after": null,
  "declaration-bang-space-before": null,
  "declaration-block-no-duplicate-custom-properties": [
    true,
  ],
  "declaration-block-no-duplicate-properties": [
    true,
    {
      "ignore": [
        "consecutive-duplicates-with-different-values",
      ],
    },
  ],
  "declaration-block-no-redundant-longhand-properties": null,
  "declaration-block-no-shorthand-property-overrides": [
    true,
  ],
  "declaration-block-semicolon-newline-after": null,
  "declaration-block-semicolon-newline-before": null,
  "declaration-block-semicolon-space-after": null,
  "declaration-block-semicolon-space-before": null,
  "declaration-block-single-line-max-declarations": [
    1,
  ],
  "declaration-block-trailing-semicolon": null,
  "declaration-colon-newline-after": null,
  "declaration-colon-space-after": null,
  "declaration-colon-space-before": null,
  "declaration-empty-line-before": null,
  "declaration-no-important": [
    true,
  ],
  "declaration-property-value-disallowed-list": [
    {
      "text-align": [
        "justify",
      ],
    },
  ],
  "font-family-name-quotes": [
    "always-where-recommended",
  ],
  "font-family-no-duplicate-names": [
    true,
  ],
  "font-family-no-missing-generic-family-keyword": [
    true,
  ],
  "function-calc-no-unspaced-operator": [
    true,
  ],
  "function-comma-newline-after": null,
  "function-comma-newline-before": null,
  "function-comma-space-after": null,
  "function-comma-space-before": null,
  "function-linear-gradient-no-nonstandard-direction": [
    true,
  ],
  "function-max-empty-lines": null,
  "function-name-case": [
    "lower",
  ],
  "function-parentheses-newline-inside": null,
  "function-parentheses-space-inside": null,
  "function-url-quotes": [
    "always",
  ],
  "function-whitespace-after": [
    "always",
  ],
  "hue-degree-notation": [
    "angle",
  ],
  "indentation": null,
  "keyframe-declaration-no-important": [
    true,
  ],
  "keyframes-name-pattern": null,
  "length-zero-no-unit": [
    true,
  ],
  "linebreaks": null,
  "max-empty-lines": null,
  "max-line-length": null,
  "max-nesting-depth": [
    4,
  ],
  "media-feature-colon-space-after": null,
  "media-feature-colon-space-before": null,
  "media-feature-name-case": null,
  "media-feature-name-no-unknown": [
    true,
  ],
  "media-feature-name-no-vendor-prefix": [
    true,
  ],
  "media-feature-parentheses-space-inside": null,
  "media-feature-range-operator-space-after": [
    "always",
  ],
  "media-feature-range-operator-space-before": [
    "always",
  ],
  "media-query-list-comma-newline-after": null,
  "media-query-list-comma-newline-before": null,
  "media-query-list-comma-space-after": null,
  "media-query-list-comma-space-before": null,
  "named-grid-areas-no-invalid": [
    true,
  ],
  "no-descending-specificity": null,
  "no-duplicate-at-import-rules": [
    true,
  ],
  "no-duplicate-selectors": [
    true,
  ],
  "no-empty-first-line": null,
  "no-empty-source": [
    true,
  ],
  "no-eol-whitespace": null,
  "no-extra-semicolons": null,
  "no-invalid-double-slash-comments": [
    true,
  ],
  "no-invalid-position-at-import-rule": [
    true,
  ],
  "no-irregular-whitespace": [
    true,
  ],
  "no-missing-end-of-source-newline": null,
  "number-leading-zero": [
    "always",
  ],
  "number-max-precision": [
    4,
  ],
  "number-no-trailing-zeros": null,
  "order/order": [
    [
      "dollar-variables",
      "custom-properties",
      {
        "hasBlock": false,
        "type": "at-rule",
      },
      "declarations",
    ],
  ],
  "property-case": null,
  "property-no-unknown": [
    true,
  ],
  "property-no-vendor-prefix": [
    true,
  ],
  "rule-empty-line-before": [
    "always-multi-line",
    {
      "except": [
        "first-nested",
      ],
      "ignore": [
        "after-comment",
      ],
    },
  ],
  "scale-unlimited/declaration-strict-value": [
    [
      "color",
      "fill",
      "stroke",
      "/-color/",
    ],
    {
      "ignoreKeywords": [
        "currentColor",
        "inherit",
        "transparent",
        "initial",
        "none",
        "unset",
        "Canvas",
        "CanvasText",
        "LinkText",
        "VisitedText",
        "ActiveText",
        "ButtonFace",
        "ButtonText",
        "ButtonBorder",
        "Field",
        "FieldText",
        "Highlight",
        "HighlightText",
        "SelectedItem",
        "SelectedItemText",
        "Mark",
        "MarkText",
        "GrayText",
        "AccentColor",
        "AccentColorText",
      ],
    },
  ],
  "scss/at-mixin-argumentless-call-parentheses": [
    "always",
  ],
  "scss/at-mixin-pattern": [
    /\\^\\[a-z0-9-\\]\\+\\$/,
  ],
  "scss/at-rule-no-unknown": [
    true,
    {
      "ignoreAtRules": [
        "tailwind",
        "apply",
        "variants",
        "responsive",
        "screen",
        "layer",
      ],
    },
  ],
  "scss/declaration-nested-properties": [
    "never",
  ],
  "scss/dollar-variable-no-missing-interpolation": [
    true,
  ],
  "scss/percent-placeholder-pattern": [
    "^do-not-use-placeholders$",
  ],
  "scss/selector-no-redundant-nesting-selector": [
    true,
  ],
  "selector-attribute-brackets-space-inside": null,
  "selector-attribute-operator-space-after": null,
  "selector-attribute-operator-space-before": null,
  "selector-attribute-quotes": null,
  "selector-class-pattern": [
    /\\^\\[a-z\\]\\+\\(-\\[a-z0-9\\]\\+\\)\\*\\(__\\[a-z0-9\\]\\+\\(-\\[a-z0-9\\]\\+\\)\\*\\)\\?\\(--\\[a-z0-9\\]\\+\\(-\\[a-z0-9\\]\\+\\)\\*\\)\\?\\$/,
    {
      "resolveNestedSelectors": true,
    },
  ],
  "selector-combinator-space-after": null,
  "selector-combinator-space-before": null,
  "selector-descendant-combinator-no-non-space": null,
  "selector-id-pattern": null,
  "selector-list-comma-newline-after": null,
  "selector-list-comma-newline-before": null,
  "selector-list-comma-space-after": null,
  "selector-list-comma-space-before": null,
  "selector-max-empty-lines": null,
  "selector-max-id": [
    0,
  ],
  "selector-max-specificity": [
    "0,4,0",
  ],
  "selector-max-type": [
    2,
    {
      "ignore": [
        "child",
        "compounded",
        "next-sibling",
      ],
    },
  ],
  "selector-no-vendor-prefix": [
    true,
  ],
  "selector-pseudo-class-case": null,
  "selector-pseudo-class-no-unknown": [
    true,
  ],
  "selector-pseudo-class-parentheses-space-inside": null,
  "selector-pseudo-element-case": null,
  "selector-pseudo-element-colon-notation": [
    "double",
  ],
  "selector-pseudo-element-no-unknown": [
    true,
  ],
  "selector-type-case": [
    "lower",
  ],
  "selector-type-no-unknown": [
    true,
    {
      "ignore": [
        "custom-elements",
      ],
    },
  ],
  "shorthand-property-no-redundant-values": null,
  "string-no-newline": [
    true,
  ],
  "string-quotes": null,
  "unicode-bom": null,
  "unit-case": null,
  "unit-no-unknown": [
    true,
  ],
  "value-keyword-case": [
    "lower",
  ],
  "value-list-comma-newline-after": null,
  "value-list-comma-newline-before": null,
  "value-list-comma-space-after": null,
  "value-list-comma-space-before": null,
  "value-list-max-empty-lines": null,
  "value-no-vendor-prefix": [
    true,
  ],
}
`;
